flow/genericactors.actor.cpp

```diff
@@ -136,3 +137,45 @@ ACTOR Future<Void> lowPriorityDelay(double waitTime) {
 	}
 	return Void();
 }
+
+namespace {
+
+struct DummyState {
+	int changed{ 0 };
+	int unchanged{ 0 };
+	bool operator==(DummyState const& rhs) const { return changed == rhs.changed && unchanged == rhs.unchanged; }
+	bool operator!=(DummyState const& rhs) const { return !(*this == rhs); }
+};
+
+ACTOR Future<Void> testPublisher(Reference<AsyncVar<DummyState>> input) {
+	state int i = 0;
+	for (; i < 100; ++i) {
+		wait(delay(deterministicRandom()->random01()));
+		auto var = input->get();
+		++var.changed;
+		input->set(var);
+	}
+	return Void();
+}
+
+ACTOR Future<Void> testSubscriber(Reference<IDependentAsyncVar<int>> output, Optional<int> expected) {
+	loop {
+		wait(output->onChange());
+		ASSERT(expected.present());
+		if (output->get() == expected.get()) {
+			return Void();
+		}
+	}
+}
+
+} // namespace
+
+TEST_CASE("/flow/genericactors/DependentAsyncVar") {
+	auto input = makeReference<AsyncVar<DummyState>>();
+	state Future<Void> subscriber1 =
+	    testSubscriber(IDependentAsyncVar<int>::create(input, [](auto const& var) { return var.changed; }), 100);
+	state Future<Void> subscriber2 =
+	    testSubscriber(IDependentAsyncVar<int>::create(input, [](auto const& var) { return var.unchanged; }), {});
+	wait(subscriber1 && testPublisher(input));
```

Comment:
`subscriber2` is not supposed to be triggered here? I would add an assert that the `subscriber2` future is not ready.